src/extensions/common/getRecordingInfo.ts

```diff
@@ -0,0 +1,81 @@
+import { useContext, useRef, useState } from "react";
+import { Recording, RecordingInfo, Sorting, SortingInfo } from "../extensionInterface";
+import { HitherContext, HitherInterface, HitherJob, useHitherJob } from "./hither";
+
+export const getRecordingInfo = async (a: {recordingObject: any, hither: HitherInterface}): Promise<RecordingInfo> => {
+    const recordingInfoJob = a.hither.createHitherJob(
+        'createjob_get_recording_info',
+        { recording_object: a.recordingObject },
+        {
+            useClientCache: true
+        }
+    )
+    const info = await recordingInfoJob.wait();
+    return info as RecordingInfo;
+}
+
+export const useRecordingInfo = (recordingObject: any): RecordingInfo | undefined => {
+    const {result: recordingInfo} = useHitherJob<RecordingInfo>(
```

Comment:
This custom hook returns the recordingInfo associated with a recordingObject. It looks in the client-side hither job cache to find the result. If it is not there, the job is sent to the server. If the result is not yet available, this function will return undefined.